src/formatDate.js

```diff
@@ -1,14 +1,38 @@
 'use strict';
 
-/**
- * @param {string} date
- * @param {string[]} fromFormat
- * @param {string[]} toFormat
- *
- * @returns {string}
- */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat.splice(3, 1);
+  const newSeparator = toFormat.splice(3, 1);
+  const dateSplit = date.split(separator);
+  const result = [];
+
+  for (let i = 0; i < toFormat.length; i++) {
+    if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+      if (dateSplit[fromFormat.indexOf('YY')] < 30) {
+        result[i] = `20${dateSplit[fromFormat.indexOf('YY')]}`;
+      } else {
+        result[i] = `19${dateSplit[fromFormat.indexOf('YY')]}`;
+      }
+    }
```

Comment:
The logic for converting 'YY' to 'YYYY' is correct, but ensure that the index used in `dateSplit` corresponds to the correct position of 'YY' in `fromFormat`.